flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/catalog/AbstractJdbcCatalog.java

```diff
@@ -116,6 +122,36 @@ public String getBaseUrl() {
 		return baseUrl;
 	}
 
+	// ------ Postgres default objects that shouldn't be exposed to users ------
+
+	/**
+	 * Retrieve the list of system schemas to ignore.
+	 */
+	public abstract Set<String> getBuiltinSchemas();
+
+	/**
+	 * Retrieve the list of system database to ignore.
+	 */
+	public abstract Set<String> getBuiltinDatabases();
+
+	// ------ retrieve PK constraint ------
+
+	protected Map.Entry<String, List<String>> getPrimaryKey(DatabaseMetaData metaData, PostgresTablePath pgPath) throws SQLException {
+		ResultSet rs = metaData.getPrimaryKeys(null,  pgPath.getPgSchemaName(), pgPath.getPgTableName());
+		Map.Entry<String, List<String>> ret = null;
+		while (rs.next()) {
+			String schema = rs.getString("table_schem");
+			String columnName = rs.getString("column_name");
+			String pkName = rs.getString("pk_name");
+			if (!getBuiltinSchemas().contains(schema)) {
```

Comment:
Don't need to call `getBuiltinSchemas().contains(..)`, because the `schema` is must not be in builtin shemas, it must be the schema of users's table.